docs/rfc/rfc-020-onboarding-projects.rst

```diff
@@ -0,0 +1,235 @@
+remove the legacy tests entirely once the new tests have proven themselves. 
+
+Abstract Orchestration Framework
+++++++++++++++++++++++++++++++++
+
+The orchestration of e2e test processes is presently done using docker
+compose, which works well, but has proven a bit limiting as all processes need
+to run on a single machine, and the log aggregation functions are confusing at
+best. 
+
+This project would replace the current orchestration with something more
+generic, potentially maintaining the current system, but also allowing the e2e
+tests to manage processes using k8s. There are a few "local" k8s frameworks
+(e.g. kind and k3s,) which might be able to be useful for our current testing
+model, but hopefully, we could use this new implementation with other k8s
+systems for more flexible distribute test orchestration. 
+
+Improve Operationalize of ``run-multiple.sh``
++++++++++++++++++++++++++++++++++++++++++++++
+
+The e2e test runner currently runs a single test, and in most cases we manage
+the test cases using a shell script that ensure cleanup of entire test
+suites. This is a bit difficult to maintain and makes reproduction of test
+cases more awkward than it should be. The e2e ``runner`` itself should provide
+equivalent functionality to ``run-multiple.sh``: ensure cleanup of test cases,
+collect and process output, and be able to manage entire suites of cases. 
+
+It might also be useful to implement an e2e test orchestrator that runs all
+tendermint instances in a single process, using "real" networks for faster
+feedback and iteration during development.
+
+In addition to being a bit easier to maintain, having a more capable runner
+implementation would make it easier to collect data from test runs, improve
+debugability and reporting. 
+
+Fan-Out For CI E2E Tests
+++++++++++++++++++++++++
+
+While there are some parallelism in the execution of e2e tests, each e2e test
+job must build a tendermint e2e image, which takes about 5 minutes of CPU time
+per-task, which given the size of each of the runs. 
+
+We'd like to be able to reduce the amount of overhead per-e2e tests while
+keeping the cycle time for working with the run a lot of different e2e
+configurations,and also have. While this is in some ways an impossible
+tradeoff, the percentage of overhead at the moment is large enough that we can
+make some material progress with a moderate amount of time. 
+
+Most of this work has to do with modifying github actions configuration and
+e2e artifact (docker) building to reduce redundant work. Eventually, when we
+can drop the requirement for CGo storage engines, it will be possible to move
+(cross) compile tendermint locally, and then inject the binary into the docker
+container, which would reduce a lot of the build-time complexity, although we
+can move more in this direction or have runtime flags to disable CGo
+dependencies for local development.
+
+Remove Panics
```

Comment:
This is simultaneously smaller and larger than it appears on the surface.

A lot of the more obvious panics we've already gotten rid of. Many of those that remain (perhaps most) are likely to require deeper changes if we really want to get rid of them. Some we may want to leave alone and just document better.

I don't disagree this is a reasonable project, but I'm not sure how to scope it.

Reply:
Yeah, maybe "remove (some) panics". I think there are still a lot that probably ought to still be gotten rid of, and it might be an instructive exercise. I think we should present it as a "here are five panics that you should delete" and have them go about it like that.